framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -387,7 +387,7 @@ void checkRep(String aliasName) {
     /** Mapping from a Tree to its TreePath. Shared between all instances. */
```

Comment:
In AnnotatedTypeFactory, there is code that copies the artificial trees from one subchecker to another.  I think you can remove this code now:
```
    public final void setEnclosingElementForArtificialTree(Tree tree, Element enclosing) {
        artificialTreeToEnclosingElementMap.put(tree, enclosing);
        for (BaseTypeChecker checker : checker.getSubcheckers()) {
            AnnotatedTypeFactory subFactory = checker.getTypeFactory();
            subFactory.artificialTreeToEnclosingElementMap.putAll(
                    artificialTreeToEnclosingElementMap);
        }
    }
```